extensions/wikia/PremiumPageHeader/PremiumPageHeaderController.class.php

```diff
@@ -13,13 +66,239 @@ public function wikiHeader() {
+		}
+
+		// items to be added to "edit" dropdown
+		$actions = array_merge( $editActions,
+			[ 'history', 'move', 'protect', 'unprotect', 'delete', 'undelete', 'replace-file' ] );
+
+		// Enable to modify actions list on dropdown
+		wfRunHooks( 'PageHeaderDropdownActions', [ &$actions ] );
+
+		foreach ( $actions as $action ) {
+			if ( isset( $this->content_actions[$action] ) ) {
+				$ret[$action] = $this->content_actions[$action];
+			}
+		}
+
+		return $ret;
+	}
+
+	/**
+	 * Copied from CommentsLikesController.class.php
+	 * Are article comments enabled for context title?
+	 */
+	private function checkArticleComments() {
+		global $wgTitle;
+		$this->isArticleComments = class_exists( 'ArticleComment' ) && ArticleCommentInit::ArticleCommentCheckTitle( $wgTitle );
+		return $this->isArticleComments;
+	}
+
+	/**
+	 * Get URL of the page comments button should be linking to
+	 */
+	private function getCommentsLink() {
+		wfProfileIn( __METHOD__ );
+		global $wgTitle, $wgRequest;
+
+		$isHistory = $wgRequest->getVal( 'action' ) == 'history';
+
+		if ( $this->checkArticleComments() ) {
+			// link to article comments section
+			if ( $wgTitle != $wgTitle || $isHistory ) {
+				$commentsLink = $wgTitle->getLocalUrl() . '#WikiaArticleComments';
+			} else {
+				// fix for redirected articles
+				$commentsLink = '#WikiaArticleComments';
+			}
+		} else {
+			// link to talk page
+			if ( $wgTitle->canTalk( $wgTitle->getNamespace() ) ) {
+				$commentsLink = $wgTitle->getTalkPage()->getLocalUrl();
+			} else {
+				// This case shouldn't happen other than Special:ThemeDesignerPreview
+				// We're faking some comments to show a user what an article would look like
+				$commentsLink = '';
+			}
+		}
+
+		wfProfileOut( __METHOD__ );
+		return $commentsLink;
+	}
+
+	private function getEditMainPage() {
+		global $wgEnableCuratedContentExt;
+
+		if ( !empty( $wgEnableCuratedContentExt ) && CuratedContentHelper::shouldDisplayToolButton() ) {
+			return [
+				'href' => '/main/edit?useskin=wikiamobile',
+				'text' => wfMessage( 'wikiacuratedcontent-edit-mobile-main-page' )->text(),
+				'id' => 'PremiumPageHeaderCuratedContentTool'
+			];
+		}
+		return [];
+	}
+
+	private function widgetLanguages() {
+		global $wgContLanguageCode, $wgTitle;
+
+		$request_language_urls = $this->request->getVal( 'request_language_urls' );
+		if ( !empty( $request_language_urls ) ) {
+			$this->language_urls = $request_language_urls;
+		}
+
+		$this->currentLangName = Language::getLanguageName( $wgContLanguageCode );
+
+		$language_urls = $this->language_urls;
+
+		$language_urls[] = [
+			'href' => $wgTitle->getFullURL(),
+			'text' => $this->currentLangName,
+			'class' => "interwiki-{$wgContLanguageCode}",
+		];
+
+		$langSortBy = [];
+
+		// only display the interlang links if there are interlanguage links
+		if ( !empty( $language_urls ) && is_array( $language_urls ) ) {
+			$lang_index = [];
+
+			// language order
+			$langSortBy = [
```

Comment:
I wonder if we should sort languages like this or maybe in alphabetical order? question probably for our designers